stdlib/src/random/random.mojo

```diff
@@ -201,3 +201,28 @@ fn randn[
     for i in range(size):
         ptr[i] = randn_float64(mean, variance).cast[type]()
     return
+
+
+fn shuffle[T: CollectionElement](inout list_in: List[T]):
+    """Shuffles the elements of the list randomly.
+
+    Performs an in-place Fisher-Yates shuffle on the provided list.
+
+    Args:
+        list_in: The list to modify.
+
+    Parameters:
+        T: The type of element in the List.
+    """
+
+    var length = len(list_in)
+
+    for i in range(length - 1, 0, -1):
+        var j: Int = random_ui64(0, i).cast[DType.int64]().value
+
+        var i_ptr = list_in.data.offset(i)
```

Comment:
I think the `offset` method is introduced for compatibility, we'd better use `+` instead.

Reply:
Ah, I was wondering the discrepancy between `+` and `offset`. Thank you.

Reply:
I actually prefer `offset`, but Connor suggested `+` in this [comment](https://github.com/modularml/mojo/pull/3203#discussion_r1672630293).